### ISSUE TITLE
Add configuration for delay (ms).

### ISSUE THREAD

**Issue by jung-hunsoo** [phoenixframework/phoenix_live_reload#68]
I found that sometimes live_reload is triggered before the codes being compiled. (on ubuntu 14.04)

Of course the best solution is to track the changed file properly, but it'll be an useful walkaround to configure delay (in milliseconds) before triggering reload.

And for the default value, I think 1~2s will be ok in general dev env.

**Comment by josevalim** [comment]
@jung-hunsoo thanks! Before we discuss adding this to Phoenix per se, can you verify that it actually works for you? You can try the following:

1. Open up deps/phoenix_live_reload and change the channel to have a `Process.sleep(1000)` before pushing the event.

2. Then run `mix deps.compile phoenix_live_reload`

3. Then try it out and let us know if it solves the issues you are seeing. :)

**Comment by jung-hunsoo** [comment]
@josevalim Thanks for help.

I found that there're two points to place a sleep function; Server side and Client side.

Though `Process.sleep(1000)` looks elixir syntax, I tried `setTimeout(function(){ reloadStrategy(chan); }, 3000);` in the `chan.on('assets_change', function(msg) {` in `priv/static/phoenix_live_reload.js` and it works.

I don't have enough experience to find out where to put the codes, so I modified js. But I think it's more reliable to trigger the event on the server side. And it'll be easier to pull it out the delay value to the configuration file.

I'll dig into and update here.

**Comment by jung-hunsoo** [comment]
I added `Process.sleep(1000)` in line 24 of `lib/phoenix_live_reload/channel.ex` and it worked greatly.

Both ex and js side works properly!

**Comment by josevalim** [comment]
@jung-hunsoo using Process.sleep() in this case is fine because it will sleep only the channel this particular browser tab is talking to... but I am thinking handling it on the JS side may be enough/better. If you add a sleep only on JS land... is it enough?

**Comment by jung-hunsoo** [comment]
@josevalim Your're right, on js side should be enough. Applying to either sides work with no difference, and it looks more reliable to hold an interval on client side.

Only problem is that I don't know how to pass the delay value in configuration file to the js, due to my poor experience.

**Comment by josevalim** [comment]
I think you would need to set a variable here: https://github.com/phoenixframework/phoenix_live_reload/blob/master/lib/phoenix_live_reload/live_reloader.ex#L74

And then use this variable in the JS code. Similar to how we do with the socket variable.

**Comment by chrismccord** [comment]
I'm confused on the cause of this issue since the live reload only triggers a page refresh, which will necessarily recompile the code, so I don't see where the race condition exists? What am I missing that adding a delay fixes?

**Comment by jung-hunsoo** [comment]
@chrismccord Actually there are two issues what I'm experiencing with live reload.

1. Live reload is triggered four times per every file change. For example, when I edit and save the `show.html.eex`, `[debug] Live reload: lib/nohogu_web/templates/tips/product/show.html.eex` is printed four times. I wanted to find out why, but not done yet.

2. Sometimes when I edited and saved the template file, the browser showed a blank where a compiled template is expected. In this case I saved the file 2~3 times, then the expected page is shown. I didn't find why, only suspect that there may be a race condition which results in a blank compiled file. (Maybe if the changed file is read on compile?)

Providing a minimal delay is not a solution for these issues, but would make a walk-around with lower cost, I think.
